src/lib/label-value/label-value.ts

```diff
@@ -55,81 +53,71 @@ declare global {
  */
-@customElement({
-  name: LABEL_VALUE_CONSTANTS.elementName
-})
-export class LabelValueComponent extends BaseComponent implements ILabelValueComponent {
-  public static get observedAttributes(): string[] {
-    return Object.values(LABEL_VALUE_CONSTANTS.observedAttributes);
-  }
+@customElement(LABEL_VALUE_TAG_NAME)
+export class LabelValueComponent extends BaseLitElement implements ILabelValueComponent {
+  public static styles = unsafeCSS(styles);
+
+  /** @deprecated Used for compatibility with legacy Forge @customElement decorator. */
+  public static [CUSTOM_ELEMENT_NAME_PROPERTY] = LABEL_VALUE_TAG_NAME;
 
-  private _empty = false;
-  private _ellipsis = false;
-  private _inline = false;
+  #internals: ElementInternals;
 
   constructor() {
     super();
-    attachShadowTemplate(this, template, styles);
+    this.#internals = this.attachInternals();
   }
 
-  public attributeChangedCallback(name: string, oldValue: string, newValue: string): void {
-    switch (name) {
-      case LABEL_VALUE_CONSTANTS.observedAttributes.EMPTY:
-        this.empty = coerceBoolean(newValue);
-        break;
-      case LABEL_VALUE_CONSTANTS.observedAttributes.ELLIPSIS:
-        this.ellipsis = coerceBoolean(newValue);
-        break;
-      case LABEL_VALUE_CONSTANTS.observedAttributes.INLINE:
-      case LABEL_VALUE_CONSTANTS.observedAttributes.DENSE:
-        this.inline = coerceBoolean(newValue);
-        break;
-    }
-  }
+  /** Controls whether the value is displayed in an alternative emphasized style. */
+  @property({ type: Boolean, reflect: true })
+  public empty = false;
 
-  public get empty(): boolean {
-    return this._empty;
-  }
-  public set empty(value: boolean) {
-    value = Boolean(value);
-    if (this._empty !== value) {
-      this._empty = value;
-      this.toggleAttribute(LABEL_VALUE_CONSTANTS.attributes.EMPTY, this._empty);
-    }
-  }
+  /** Controls whether overflowing text is truncated with an ellipsis. */
+  @property({ type: Boolean, reflect: true })
+  public ellipsis = false;
 
-  public get ellipsis(): boolean {
-    return this._ellipsis;
-  }
-  public set ellipsis(value: boolean) {
-    value = Boolean(value);
-    if (this._ellipsis !== value) {
-      this._ellipsis = value;
-      this.toggleAttribute(LABEL_VALUE_CONSTANTS.attributes.ELLIPSIS, this._ellipsis);
+  /** Controls whether the label and value are displayed inline. */
+  @property({ type: Boolean, reflect: true })
+  public inline = false;
+
+  /** Deprecated. Use `inline` instead. */
+  @property({ type: Boolean, reflect: true })
+  public dense = false;
+
+  public override willUpdate(changedProperties: PropertyValues<this>): void {
+    if (changedProperties.has('dense')) {
+      if (this.hasUpdated || this.hasAttribute('dense')) {
+        // Ignore the initialization cycle. On subsequent updates, map dense to inline
+        if (this.inline !== this.dense) {
+          this.inline = this.dense;
+        }
+      }
```

Comment:
I know it's unfortunate that we have to keep these properties in sync for now to avoid breaking changes, but I'm not sure how I feel about this block. I also noticed that the dense attribute does not get removed if setting inline to false. Not really a big deal since typically only one of these properties is in use at a time, but better to handle it gracefully. Maybe we can clean this up and make it more simple/readable?